src/viser/_message_api.py

```diff
@@ -625,8 +634,41 @@ def _handle_click_updates(
         if handle is None or handle._impl.click_cb is None:
             return
         for cb in handle._impl.click_cb:
-            event = ClickEvent(client_id=client_id, target=handle)
+            event = SceneNodePointerEvent(client_id=client_id, target=handle)
             cb(event)  # type: ignore
+    
+    @property
+    def scene_pointer_enabled(self) -> bool:
+        """Whether scene pointer events are enabled."""
+        return self._scene_pointer_enabled
+
+    @scene_pointer_enabled.setter
+    def scene_pointer_enabled(self, enable: bool) -> None:
+        """Enable or disable scene pointer events."""
+        self._scene_pointer_enabled = enable
+        self._queue(_messages.EnableScenePointerMessage(
+            enabled=enable
+            ))
+
+    def _handle_scene_pointer_updates(
+        self, client_id: ClientId, message: _messages.ScenePointerMessage
+    ):
+        """Callback for handling click messages."""
+        for cb in self._scene_pointer_cb:
+            event = ScenePointerEvent(
+                client_id=client_id,
+                pointer_type=message.pointer_type,
+                ray_origin=message.ray_origin,
+                ray_direction=message.ray_direction,
+            )
+            cb(event)
+
+    def on_scene_pointer(
```

Comment:
maybe `on_scene_click()` or `on_scene_pointer_event()`?

where with the former we would later add `on_scene_double_click()` etc

I have a mild preference for the latter because simplicity, but it might be less efficient because with the former we can just toggle only the events that we're actually listening to